### PR TITLE
First pass - optimizing code search and event queries

### PR DESCRIPTION
**Pull request by emmanueldenloye**

### CODE REVIEW

File: exec/Main.hs [hunk 1]
```diff
@@ -89,6 +95,21 @@ main = do
       Args _ _ _ _ level _ -> level
       RichListArgs _ level -> level
 
+addTransactionsHeightIndex :: LogFunctionIO Text -> Connection -> IO ()
```
Comment (mightybyte) [inline exec/Main.hs:98]: We should probably make these functions check to see whether the index exists and skip the `create` if it does.

**Review by mightybyte — CHANGES_REQUESTED** [review]
The cabal.project file got deleted.  We need to revert that.

File: lib/ChainwebDb/Types/Transaction.hs [hunk 2]
```diff
@@ -88,6 +92,9 @@ Transaction
 type Transaction = TransactionT Identity
 type TransactionId = PrimaryKey TransactionT Identity
 
+instance HasColumnType TsVector where
```
Comment (mightybyte) [inline lib/ChainwebDb/Types/Transaction.hs:95]: I think this might be no longer needed, right?

File: lib/ChainwebDb/Types/Transaction.hs [hunk 3]
```diff
@@ -7,6 +7,7 @@
 {-# LANGUAGE MultiParamTypeClasses #-}
 {-# LANGUAGE NoImplicitPrelude #-}
 {-# LANGUAGE TypeFamilies #-}
+{-# LANGUAGE TypeApplications #-}
```
Comment (mightybyte) [inline lib/ChainwebDb/Types/Transaction.hs:10]: If the below question is "yes", then this is probably unnecessary as well.

**Review by mightybyte — CHANGES_REQUESTED** [review]

File: lib/ChainwebDb/Types/Transaction.hs (hunk 2 above)
Comment (emmanueldenloye) [reply lib/ChainwebDb/Types/Transaction.hs:95]: This can be removed.

**Review by mightybyte — APPROVED** [review]

**Pull request merged**